tasks/Task.cpp

```diff
@@ -13,37 +47,650 @@ Task::~Task()
 {
 }
 
-
-
 /// The following lines are template definitions for the various state machine
 // hooks defined by Orocos::RTT. See Task.hpp for more detailed
 // documentation about them.
 
 bool Task::configureHook()
 {
-    if (! TaskBase::configureHook())
+    if (!TaskBase::configureHook()) {
         return false;
+    }
+
+    try {
+        m_system = Arena::OpenSystem();
+        connectToCamera();
+        switchOverAccess();
+        configureCamera();
+    }
+    catch (GenICam::GenericException& ge) {
+        LOG_ERROR_S << "GenICam exception thrown: " << ge.what() << endl;
+        throw std::runtime_error(ge.what());
+    }
     return true;
 }
+
 bool Task::startHook()
 {
-    if (! TaskBase::startHook())
+    if (!TaskBase::startHook()) {
         return false;
+    }
+
+    LOG_INFO_S << "Starting frame acquisition." << endl;
+    try {
+
+        m_device->StartStream();
+    }
+    catch (GenICam::GenericException& ge) {
+        LOG_ERROR_S << "GenICam exception thrown: " << ge.what() << endl;
+        throw std::runtime_error(ge.what());
+    }
     return true;
 }
 void Task::updateHook()
 {
     TaskBase::updateHook();
+    acquireFrame();
 }
 void Task::errorHook()
 {
     TaskBase::errorHook();
+    try {
+        m_device->StopStream();
+    }
+    catch (GenICam::GenericException& ge) {
+        LOG_ERROR_S << "GenICam exception thrown: " << ge.what() << endl;
+        throw std::runtime_error(ge.what());
+    }
```

Comment:
errorHook is a runtime state, that is it exists to have a "degraded" but workable state while the component is running.

A.k.a. You don't need this.